File: tezrok-api/src/main/java/io/tezrok/api/util/VelocityUtil.java
package io.tezrok.api.util;

import org.apache.velocity.Template;
import org.apache.velocity.app.VelocityEngine;
import org.apache.velocity.runtime.RuntimeConstants;
import org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader;

public final class VelocityUtil {
    public static Template getTemplate(String filePath) {
        VelocityEngine engine = new VelocityEngine();
        engine.setProperty(RuntimeConstants.RESOURCE_LOADER, "classpath");
        engine.setProperty("classpath.resource.loader.class", ClasspathResourceLoader.class.getName());
        Template template = engine.getTemplate(filePath, "UTF-8");

        return template;
    }
}

File: tezrok-core/src/main/kotlin/io/tezrok/core/util/VelocityUtil.kt
package io.tezrok.core.util

import org.apache.velocity.Template
import org.apache.velocity.app.VelocityEngine
import org.apache.velocity.runtime.RuntimeConstants
import org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader

object VelocityUtil {
    fun getTemplate(filePath: String): Template {
        val engine = VelocityEngine()
        engine.setProperty(RuntimeConstants.RESOURCE_LOADER, "classpath")
        engine.setProperty("classpath.resource.loader.class", ClasspathResourceLoader::class.java.name)

        return engine.getTemplate(filePath, "UTF-8")
    }
}
